dptb/nn/energy.py

```
@@ -113,6 +123,178 @@ def forward(self,
+            
+            A.sort_indices()
+            A.sum_duplicates()
+            N = A.shape[0]
+            
+            # Try PARDISO first, fall back to scipy SuperLU if PARDISO fails
+            # (MKL PARDISO has a known bug with certain block-structured patterns)
+            solver = PyPardisoSolver(mtype=13)
+            solver.factorize(A)
+            
+            def matvec(b):
+                return solver.solve(A, b)
```

Ruff (0.15.0)

[warning] Function definition does not bind loop variable `solver`

(B023)